### PR TITLE
Adding ability to set idle timeout in amazon elastic load balancer

### PR DESCRIPTION
**Pull request by digital-abyss**
Implements the ability to set the idle timeout as described in 
https://docs.aws.amazon.com/ElasticLoadBalancing/latest/DeveloperGuide/config-idle-timeout.html?console_help=true

### CODE REVIEW

**Comment by gregdek** [comment]
@jsdalton @rickmendes Please review this PR to make sure it adheres to the following guidelines:

http://docs.ansible.com/developing_modules.html#module-checklist

If it passes these guidelines, and if you believe it’s a good PR otherwise, please add a comment with "shipit" in the text, and we will flag it for inclusion.

If it fails these guidelines, please add a comment with "needs_revision" in the text, along with the needed changes, and we will flag it for revision.

File: cloud/amazon/ec2_elb_lb.py [hunk 1]
```diff
@@ -708,6 +719,12 @@ def _set_connection_draining_timeout(self):
             attributes.connection_draining.enabled = False
             self.elb_conn.modify_lb_attribute(self.name, 'ConnectionDraining', attributes.connection_draining)
 
+    def _set_idle_timeout(self):
+        attributes = self.elb.get_attributes()
+        if self.idle_timeout is not None and self.idle_timeout != attributes.connecting_settings.idle_timeout:
+            attributes.connecting_settings.idle_timeout = self.idle_timeout
+            self.elb_conn.modify_lb_attribute(self.name, 'connectingsettings',attributes.connecting_settings)
```
Comment (jsdalton) [inline cloud/amazon/ec2_elb_lb.py]: Minor formatting issue: Add space between `'connectingSettings',` and `attributes`

**Comment by jsdalton** [comment]
Fix the missing space and all else looks :+1:

**Comment by digital-abyss** [comment]
@jsdalton  -- I believe I've fixed up the commit with the extra space.  The Travis CI build failed, but not sure if it is another issue ?

```
0.12s$ python2.4 -m compileall -fq -x 'cloud/' .

Compiling ./system/cron.py ...

  File "./system/cron.py", line 241

    os.chmod(path, 0o644)

                       ^

SyntaxError: invalid syntax
```

File: cloud/amazon/ec2_elb_lb.py [hunk 2]
```diff
@@ -708,6 +719,12 @@ def _set_connection_draining_timeout(self):
             attributes.connection_draining.enabled = False
             self.elb_conn.modify_lb_attribute(self.name, 'ConnectionDraining', attributes.connection_draining)
 
+    def _set_idle_timeout(self):
+        attributes = self.elb.get_attributes()
+        if self.idle_timeout is not None and self.idle_timeout != attributes.connecting_settings.idle_timeout:
```
Comment (ghost) [inline cloud/amazon/ec2_elb_lb.py:723]: not None never reads well to me. How about:

``` python
if self.idle_timeout is defined and self.idle_timeout != attributes.connecting_settings.idle_timeout:
```

File: cloud/amazon/ec2_elb_lb.py [hunk 3]
```diff
@@ -107,6 +107,12 @@
     required: false
     aliases: []
     version_added: "1.8"
+  idle_timeout:
+     description:
+       - Sets the number of seconds a connection can be idle before the load balancer terminates the connection
+     required: false
+     aliases: []
```
Comment (ghost) [inline cloud/amazon/ec2_elb_lb.py]: not needed when empty

Comment (digital-abyss) [inline cloud/amazon/ec2_elb_lb.py]: @rickmendes -- I'm not sure what you mean, should I remove this ?

Comment (ghost) [inline cloud/amazon/ec2_elb_lb.py]: @digital-abyss Yes. If you look at the link provided by @gregdek, the module checklist calls these out as not being necessary when there are no aliases. I know the existing code had them. However, it is my opinion that new code should not ignore the guidelines just to be consistent with the old code. We will never have code that passes the checklist if we don't reverse our old habits as we add new code.

File: cloud/amazon/ec2_elb_lb.py (hunk 2 above)
Comment (digital-abyss) [inline cloud/amazon/ec2_elb_lb.py:723]: I was following the convention from within _set_connection_raining_timeout(self):

``` python
 if self.connection_draining_timeout is not None:
```

What convention should I use for checking for defined ?  I could try something like:

``` python
try:
    if self.idle_timeout != attributes.connecting_settings.idle_timeout:
        attributes.connecting_settings.idle_timeout = self.idle_timeout
        self.elb_conn.modify_lb_attribute(self.name, 'connectingsettings', attributes.connecting_settings)
except NameError:
    pass
```

File: cloud/amazon/ec2_elb_lb.py (hunk 3 above)
Comment (digital-abyss) [inline cloud/amazon/ec2_elb_lb.py]: Sounds good.  Will update the PR.

File: cloud/amazon/ec2_elb_lb.py (hunk 2 above)
Comment (ghost) [inline cloud/amazon/ec2_elb_lb.py:723]: @digital-abyss This is my bad. I always think not None looks bad until I see the workarounds to check if a variable is defined. I am not sure except NameError or if None else are really cleaner. I should take this up with the Python gods. If Ansible can have is defined, why can't Python have is_defined()?

So don't stress over this. You can leave it as is.

**Comment by digital-abyss** [comment]
@rickmendes @jsdalton -- please let me know if there are any other changes/issues.  Thank you :)

**Comment by gregdek** [comment]
Thanks @digital-abyss for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by robynbergeron** [comment]
@digital-abyss A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by digital-abyss** [comment]
@robynbergeron : it appears @abridgett duplicated the contents of this pull request with https://github.com/ansible/ansible-modules-core/pull/2228, which has been merged in.

**Pull request closed without merging**